modules/customize-certificates-add-service-serving.adoc

```diff
@@ -5,15 +5,16 @@
 [id="add-service-certificate_{context}"]
 = Add a service certificate
 
-To secure communication to your service, generate a
-signed serving certificate and key pair into a secret in the same
-namespace as the service.
+To secure communication to your service, generate a signed serving certificate and key pair into a secret in the same namespace as the service.
+
+The generated certificate is only valid for the internal service DNS name `<service.name>.<service.namespace>.svc`, and is only valid for internal communications. If your service is a headless service (no `clusterIP` value set), the generated certificate also contains a wildcard subject in the format of `*.<service.name>.<service.namespace>.svc`.
```

Comment:
(The certificates are actually valid also for `<service.name>.<service.namespace>.svc.cluster.local` (also was the case before), and newly headless services for `*.<service.name>.<service.namespace>.svc.cluster.local`. The `.cluster.local` suffix is hard-coded in service-ca, and AFAICS similarly hard-coded by the installer and MCO.

I don’t think this new feature is a reason to start documenting the `.cluster.local` subjects (and I really don’t know whether we should document them) — just noting a nit vs. the “only valid” wording.)

Reply:
… Actually we’ve discussed this earlier and leaned towards not advertising `.cluster.local` as long as it’s unclear whether the cluster domain should be user-configurable.